python/paddle/v2/framework/tests/test_dynamic_recurrent_op.py

```diff
@@ -107,5 +111,58 @@ def test_forward(self):
         print 'pd_output', pd_output
 
 
+class RecurrentGradientOpTest(unittest.TestCase):
+    py = PyRNNStep()
+
+    def create_forward_op(self):
+        # create RNNOp
+        self.forward_op = DynamicRecurrentOp(
+            # inputs
+            inlinks=["x"],
+            boot_memories=["h_boot"],
+            step_net="stepnet",
+            # outputs
+            outlinks=["h@mem"],
+            step_scopes="step_scopes",
+            # attributes
+            pre_memories=["h@pre"],
+            memories=["h@mem"])
+
+    def create_gradient_op(self):
+        a = set()
+        backward_op = core.DynamicRecurrentOp.backward(self.forward_op, a)
+
+    def create_step_net(self):
```

Comment:
We are not going to do NetOp, but block, here the step is not represented by a NetOp instance, but a block. We mustn't go on using the name "step_*net*".

Reply:
Currently, these codes still focus on the primary logic of RNN its own. This implementation is based on `NetOp` because that is stable and simpler.

I will start the migration to `block` and `executor` on next Monday.